server/memstore.go

```diff
@@ -879,6 +905,14 @@ func (ms *memStore) LoadLastMsg(subject string, smp *StoreMsg) (*StoreMsg, error
 		smp = new(StoreMsg)
 	}
 	sm.copy(smp)
+
+	// Decompress the message if needed.
+	if c, err := ms.cfg.Compression.Decompress(smp.msg); err != nil {
```

Comment:
We call into this function for every load call, even if not compressed?

Reply:
If not compressing then the slice just gets returned unmodified, so it stops us needing extra checks at the call site.